src/bin/julialauncher.rs

```diff
@@ -175,6 +175,32 @@ fn get_julia_path_from_channel(
     juliaupconfig_path: &Path,
     juliaup_channel_source: JuliaupChannelSource,
 ) -> Result<(PathBuf, Vec<String>)> {
+    get_julia_path_from_channel_impl(
+        versions_db,
+        config_data,
+        channel,
+        juliaupconfig_path,
+        juliaup_channel_source,
+        &mut std::collections::HashSet::new(),
+    )
+}
+
+fn get_julia_path_from_channel_impl(
+    versions_db: &JuliaupVersionDB,
+    config_data: &JuliaupConfig,
+    channel: &str,
+    juliaupconfig_path: &Path,
+    juliaup_channel_source: JuliaupChannelSource,
+    visited: &mut std::collections::HashSet<String>,
+) -> Result<(PathBuf, Vec<String>)> {
+    // Check for circular references
+    if visited.contains(channel) {
+        return Err(anyhow!(
+            "Circular alias detected: alias chain contains a cycle involving '{}'",
+            channel
+        ));
+    }
+    visited.insert(channel.to_string());
     let channel_valid = is_valid_channel(versions_db, &channel.to_string())?;
```

Comment:
```suggestion
    let channel_valid = is_valid_channel(versions_db, channel)?;
```
(Not part of PR, but noticed it)

Reply:
This change also causes compilation failure for me.